_posts/2016-11-22-files-cpp.md

```diff
@@ -0,0 +1,162 @@
+---
+layout: post
+title: OME Files C++ and C++11
+categories: file-formats ome-files
+---
+
+This blog is an update about the use of
+[C++11](https://en.wikipedia.org/wiki/C%2B%2B11) and
+[C++14](https://en.wikipedia.org/wiki/C%2B%2B14) ("modern C++") within
+OME Files C++.
+
+We have been evaluating the switch from C++98 to C++11 for over 18
+months.  Up until now, we have remained using C++98 in order to
+continue to support older systems with compilers which do not support
+C++11.  However, circumstances have recently changed which will
+necessitate a switch at some point in the near future.
+
+## Platform support
+
+Most current platforms support C++11 and C++14, with some exceptions
+(workarounds are noted)
+
+| Platform           | Compiler        | C++11                   | C++14                   |
+| ------------------ | --------------- | ----------------------- | ----------------------- |
+| CentOS 6           | GCC 4.6         | Unsupported by default* | Unsupported by default* |
+| CentOS 7           | GCC 4.8         | Mostly supported        | Partially supported     |
+| FreeBSD 10         | Clang/LLVM 3.4  | Fully supported         | Fully supported         |
+| FreeBSD 11         | Clang/LLVM 3.8  | Fully supported         | Fully supported         |
+| MacOS X 10.10      | Apple LLVM 7.3† | Fully supported         | Fully supported         |
+| MacOS X 10.11      | Apple LLVM 8.0‡ | Fully supported         | Fully supported         |
+| MacOS X 10.12      | Apple LLVM 8.0‡ | Fully supported         | Fully supported         |
+| Ubuntu 14.04       | GCC 4.8         | Mostly supported        | Partially supported     |
+| Ubuntu 16.04       | GCC 5.3         | Fully supported         | Fully supported§        |
+| Visual Studio 2013 | VC12‖           | Partially supported     | Mostly unsupported      |
+| Visual Studio 2015 | VC13‖           | Supported               | Partially supported     |
+| Visual Studio 2017 | VC15‖           | Supported               | Supported               |
+
+\* The software collections
+  [devtoolset-4](https://www.softwarecollections.org/en/scls/rhscl/devtoolset-4/)
+  (GCC 5.2) and
+  [devtoolset-3](https://www.softwarecollections.org/en/scls/rhscl/devtoolset-3/)
+  (GCC 4.9) packages provide updated compilers which can build OME
+  Files C++ using C++11
+
+† Probably equivalent to Clang/LLVM 3.6 (it's a proprietary fork, so
```

Comment:
Of course it does. I just don't think we need to put ourselves in a position of provoking ire in our general direction by getting caught up making statements on the proprietary vs. not nature of what Apple is doing. As we're kind of committed to supporting the platform, it's not really relevant anyway as we're going to have to decide which features to use based on testing and experience.